pkg/artifactory/resource/configuration/resource_artifactory_property_set.go

```diff
@@ -0,0 +1,290 @@
+			),
+		)
+
+		return diag.FromErr(pkr(&matchedPropertySet, d))
+	}
+
+	var parsePredefinedValues = func(values []PredefinedValue) map[string]interface{} {
+		parsedPredefinedValues := map[string]interface{}{}
+		for _, value := range values {
+			parsedPredefinedValues[value.Name] = map[string]interface{}{
+				"defaultValue": value.DefaultValue,
+			}
+		}
+		return parsedPredefinedValues
+	}
+
+	var parseProperties = func(properties []Property) map[string]interface{} {
+		parsedProperties := map[string]interface{}{}
+
+		for _, property := range properties {
+			parsedProperties[property.Name] = map[string]interface{}{
+				"predefinedValues":       parsePredefinedValues(property.PredefinedValues),
+				"closedPredefinedValues": property.ClosedPredefinedValue,
+				"multipleChoice":         property.MultipleChoice,
+			}
+		}
+		return parsedProperties
+	}
+
+	var resourcePropertySetsUpdate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		unpackedPropertySet := unpackPropertySet(d)
+
+		///* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		//There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		//GET call structure has "propertySets -> propertySet -> Array of property sets".
+		//PATCH call structure has "propertySets -> propertySet (dynamic sting). Property name and predefinedValues names are also dynamic strings".
+		//Following nested map of string structs are constructed to match the usage of PATCH call with the consideration of dynamic strings.
+		//*/
+		var constructBody = map[string]map[string]map[string]interface{}{
+			"propertySets": {
+				unpackedPropertySet.Name: {
+					"visible":    unpackedPropertySet.Visible,
+					"properties": parseProperties(unpackedPropertySet.Properties),
+				},
+			},
+		}
+
+		content, err := yaml.Marshal(&constructBody)
+
+		if err != nil {
+			return diag.Errorf("failed to marshal property set during Update")
+		}
+
+		err = SendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+		}
+
+		d.SetId(unpackedPropertySet.Name)
+		return resourcePropertySetRead(ctx, d, m)
+
+	}
+
+	var resourceLdapSettingsDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		propertySetConfigs := &PropertySets{}
+		unpackedPropertySet := unpackPropertySet(d)
+
+		response, err := m.(*resty.Client).R().SetResult(&propertySetConfigs).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		if response.IsError() {
+			return diag.Errorf("got error response for API: /artifactory/api/system/configuration request during Read")
+		}
+		var matchedPropertySet = PropertySet{} // TODO: do we match before delete or just delete based on HCL list of sets?
```

Comment:
Same issue here with usage of `var` and duplicate assignment in next line.

Reply:
Is the comment still applicable?

Reply:
Resolved.